### PR TITLE
Title should be mandatory

### LINKED ISSUE #1129: ExpressionEditor: group attributes by parent entity
in Editor attributes are seen as a long list. I would be easier to use them if they are grouped by parent entity, and they become to the list in the same order as they they are in the entry form or in table. Now items are in alphabetic order (by parent entities). 
This is a very good example what we need:
see "Grouping with Category" at https://www.syncfusion.com/javascript-ui-controls/js-dropdown-list

### LINKED ISSUE #1286: User registration: Title is mandatory

### PR DESCRIPTION
**Pull request by ramirobg94**
Resolves #1286

### CODE REVIEW

File: server/modules/user/api/userApi.js [hunk 1]
```diff
@@ -149,10 +151,16 @@ export const init = app => {
...
-
+    const body = Request.getBody(req)
+    const { user: userToUpdateString } = body
+    let userToUpdate = A.parse(userToUpdateString)
     const validation = await UserValidator.validateUser(userToUpdate)
 
+    userToUpdate = {
+      ...userToUpdate,
+      props: A.stringify(userToUpdate.props),
+    }
```
Comment (ramirobg94) [inline server/modules/user/api/userApi.js:162]: I am worried about that because due to we use formData, we should do this

Comment (minotogna) [reply server/modules/user/api/userApi.js:162]: what?

Comment (minotogna) [inline server/modules/user/api/userApi.js:162]: this is unclear. why that conversion?

**Review by minotogna — CHANGES_REQUESTED** [review]

File: webapp/views/Guest/views/ResetPassword/ResetPassword.js [hunk 2]
```diff
@@ -4,6 +4,8 @@ import React from 'react'
 import { useI18n } from '@webapp/store/system'
 import Error from '@webapp/views/Guest/Error'
 
+import DropdownUserTitle from '@webapp/views/App/views/Users/DropdownUserTitle'
```
Comment (minotogna) [inline webapp/views/Guest/views/ResetPassword/ResetPassword.js:7]: this is a common component now. move it to common components (under form)

File: webapp/views/Guest/views/ResetPassword/store/hooks.js [hunk 3]
```diff
@@ -17,24 +17,25 @@ export const useResetPassword = () => {
...
-  const { name, password, passwordConfirm } = state.user
+  const { name, password, passwordConfirm, props } = state.user
 
   const { data: { user } = {}, dispatch: getResetPasswordUser } = useAsyncGetRequest(`/auth/reset-password/${uuid}`)
 
   const {
     data: { result: resetComplete = false } = {},
     dispatch: dispatchPostResetPassword,
-  } = useAsyncPutRequest(`/auth/reset-password/${uuid}`, { name, password })
+  } = useAsyncPutRequest(`/auth/reset-password/${uuid}`, { name, props, password })
```
Comment (minotogna) [inline webapp/views/Guest/views/ResetPassword/store/hooks.js:27]: can't you just pass the title ?

File: server/modules/user/api/userApi.js (hunk 1 above)
Comment (ramirobg94) [reply server/modules/user/api/userApi.js:162]: The repository expects that the props are a string, maybe it is better to move this logic to stringify to the repository.

Comment (ramirobg94) [reply server/modules/user/api/userApi.js:162]: same as above, I felt dirty when I did that 👎

Comment (minotogna) [reply server/modules/user/api/userApi.js:162]: can u make it the same way we persist props for nodeDefs ?
This code might be old... better we fix it now.
what do you think?

Comment (ramirobg94) [reply server/modules/user/api/userApi.js:162]: Perfect! I will take a look into the nodeDef code, yep I am finding code that is not used and outdated.

**Review by minotogna — APPROVED** [review]

**Pull request merged**